server/src/com/cloud/network/element/VpcVirtualRouterElement.java

```diff
@@ -466,7 +466,7 @@ public boolean deletePrivateGateway(final PrivateGateway gateway) throws Concurr
             }
         }
 
-        return result > 0 ? true : false;
+        return result == routers.size() ? true : false;
     }
```

Comment:
@alexandrelimassantana This code is not very new, please write a little pull request to that extend. (also the issue that you quite justly pointed out is there in the old code as well and bears no weight on the validity of this PR)